sympy/sets/sets.py

```diff
@@ -534,6 +535,30 @@ def measure(self):
         """
         return self._measure
 
+    @property
+    def kind(self):
+        """
+        Finds kind of a set
```

Comment:
The docstring needs a little more explanation. Probably what would be most useful is a "see also" section linking back to other documentation about kinds. The sentence here should clarify that the kind of a set is always SetKind except parametrised by the kind of the elements of the Set.

Reply:
This will work?

        Finds kind of a set. Kind of a set will always be in SetKind,
        with kind of elements of the set inside the parenthesis,
        example
        kind of Interval(1,2) will SetKind(NumberKind)

        Examples
        ========
        
        >>> from sympy import Interval, Matrix, FiniteSet, EmptySet, ProductSet, PowerSet
        >>> FiniteSet(Matrix([1, 2])).kind
        SetKind(MatrixKind(NumberKind))
        >>> FiniteSet(0, Matrix([1, 2])).kind
        SetKind(UndefinedKind)
        >>> Interval(1,2).kind
        SetKind(NumberKind)
        >>> EmptySet.kind
        SetKind()
        >>> ProductSet(FiniteSet(1, Matrix([1, 2])), Interval(1, 2)).kind
        SetKind(TupleKind(UndefinedKind, NumberKind))
        >>> PowerSet({1,2,3}).kind
        SetKind(SetKind(NumberKind))
        
        See Also
        ========

        NumberKind

        MatrixKind